src/mqtt_client.rs

```diff
@@ -5,27 +5,35 @@ use crate::{
     Property, {debug, error, info},
 };
 
-use core::{cell::RefCell, str::FromStr};
+use core::{cell::RefCell, convert::TryFrom, str::FromStr};
 
 use embedded_nal::{nb, IpAddr, Mode, SocketAddr};
+use embedded_time::{duration::Seconds, fixed_point::FixedPoint, Clock, Instant};
 
 use generic_array::{ArrayLength, GenericArray};
 use heapless::String;
 
 /// A client for interacting with an MQTT Broker.
-pub struct MqttClient<T, N>
+pub struct MqttClient<T, N, C>
 where
     T: ArrayLength<u8>,
     N: embedded_nal::TcpStack,
+    C: Clock,
+    u32: TryFrom<C::T>,
```

Comment:
Ideally, if the user doesn't have access to a `Clock` and doesn't want the ping request/response mechanism, they still won't be able to use this library because the constructor forces them to provide one (as well as the type signature).

We could follow the route of `smoltcp` where a generic timestamp is provided in the `update` function - any thoughts here?